ludwig/utils/strings_utils.py

```diff
@@ -504,8 +505,21 @@ def build_sequence_matrix(
         logger.debug(f"max length of {format}: {max_length} < limit: {length_limit}")
     max_length = length_limit
 
+    # Set padding token id based on tokenizer_type. Huggingface tokenizers typically have a pad_token_id attribute.
+    if tokenizer_type == "hf_tokenizer":
+        if hasattr(tokenizer.tokenizer, "pad_token_id"):
```

Comment:
I think this is actually only an issue for `hf_tokenizer` objects that end up being hot-swapped for TorchText-based tokenizers, like the BERTTokenizer.

For `HFTokenizer` objects, we do set this value already: https://github.com/ludwig-ai/ludwig/blob/f829345c7df645835d50fd78d7c483ed2d12cbc7/ludwig/utils/tokenizers.py#L823

Reply:
Probably okay for now, but I would leave a TODO here for ensuring that this works with the torchtext tokenizers.